text/router-enhancements.md

```diff
@@ -110,15 +113,16 @@ The resolution order stays as it is currently. That is
 - Greater number of segments means more specific.
 - On the same number of segments, static is more specific than dynamic, and dynamic is more specific
   than globs.
-- In case of tie, the first defined route wins
+- In case of tie in the number and type of segments, the route with more handlers wins.
+- If all the previous fail, the first defined route wins
 
 Examples:
 
 ```
 /foo/bar            => 'sample1' (two segments, both static)
 /foo/dynamic        => 'sample2' (two segments, one static, one dynamic)
 /foo/something/else => 'sample3' (two segments, one static, one glob)
-/foo                => 'sample2' (one static segment)
+/foo                => 'sample4' (one static segment, but wins over sample 2 because has more handlers)
```

Comment:
This would be `sample2` because that definition creates an accepting state for `/foo` and is defined before `sample4`.